### ISSUE TITLE
Know Your Enemy 9.2.0 Incompatibility

### ISSUE THREAD

**Issue by Drunkender** [tr4wzified/KnowYourArmorPatcher#23]
Know Your Enemy version 9.2.0 compacted the form id's to ESlify the mod breaking this patcher.(It complains about missing... somthing)
Install KYE 9.1.0 Patcher runs without errors.

**Comment by selsie** [comment]
Here is the error log:

```
Compiling
Extra data folder already exists. Leaving as is: "D:\Skyrim Tools\Synthesis\Data\Skyrim Special Edition\KnowYourArmorPatcher"
("C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Git\ef00806e-260c-43ee-b77f-09231fdb9190\Runner\KnowYourArmorPatcher"): "dotnet" "build \"KnowYourArmorPatcher.csproj\" --runtime win-x64 -c Release"
Compiled
Sha "23cded3450c4bd0dfb9ff6bd1242eeadd975b739"
Running
(""): "dotnet" "run --project \"C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Git\ef00806e-260c-43ee-b77f-09231fdb9190\Runner\KnowYourArmorPatcher\KnowYourArmorPatcher.csproj\" --runtime win-x64 -c Release --no-build run-patcher --DataFolderPath \"D:\Games\steamapps\common\Skyrim Special Edition\Data\" --ExtraDataFolder \"D:\Skyrim Tools\Synthesis\Data\Skyrim Special Edition\KnowYourArmorPatcher\" --DefaultDataFolderPath C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Git\ef00806e-260c-43ee-b77f-09231fdb9190\Runner\KnowYourArmorPatcher\Data --GameRelease SkyrimSE --ModKey Synthesis3.esp --LoadOrderFilePath C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Workspace\Synthesis3\Plugins.txt --LoadOrderIncludesCreationClub --Localize --OutputPath \"C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Workspace\Synthesis3\1 - KnowYourArmorPatcher\Synthesis3.esp\" --PatcherName KnowYourArmorPatcher --UseUtf8ForEmbeddedStrings"
Mutagen version: 0.31
Mutagen sha: cf708daf0b89c643b94bf08c28e7ab2a719d9451
Synthesis version: 0.19.3
Synthesis sha: e765ef1333b34f3ded15de6ed5ec18983ee48c36
run-patcher --DataFolderPath "D:\Games\steamapps\common\Skyrim Special Edition\Data" --ExtraDataFolder "D:\Skyrim Tools\Synthesis\Data\Skyrim Special Edition\KnowYourArmorPatcher" --DefaultDataFolderPath C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Git\ef00806e-260c-43ee-b77f-09231fdb9190\Runner\KnowYourArmorPatcher\Data --GameRelease SkyrimSE --LoadOrderFilePath C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Workspace\Synthesis3\Plugins.txt --OutputPath "C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Workspace\Synthesis3\1 - KnowYourArmorPatcher\Synthesis3.esp" --PatcherName KnowYourArmorPatcher
Prepping state.

Additional Settings Files: 
 - Guards_Armor_Replacer.json - PLUGIN MISSING
 - Hothtrooper44_ArmorCompilation.json - PLUGIN MISSING

Reading settings: D:\Skyrim Tools\Synthesis\Data\Skyrim Special Edition\KnowYourArmorPatcher\settings.json
Settings Used: 
effect_intensity: 1

System.Exception: Unable to find required perk: <Skyrim.IPerkGetter>0B6D0D:know_your_enemy.esp
   at KnowYourArmorPatcher.Program.RunPatch(IPatcherState`2 state) in C:\Users\Ranbi\AppData\Local\Temp\Synthesis\2bb878d8-510a-4adb-bb95-00e8a35c3d60\Git\ef00806e-260c-43ee-b77f-09231fdb9190\Runner\KnowYourArmorPatcher\Program.cs:line 244
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.<>c__DisplayClass12_0`2.<<AddPatch>b__0>d.MoveNext() in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\SynthesisPipeline.cs:line 94
--- End of stack trace from previous location ---
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.Run(RunSynthesisMutagenPatcher args, ModKey exportKey) in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\SynthesisPipeline.cs:line 456
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.Run(RunSynthesisMutagenPatcher args) in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\SynthesisPipeline.cs:line 404
   at Mutagen.Bethesda.Synthesis.SynthesisPipeline.<>c__DisplayClass36_0.<<InternalRun>b__1>d.MoveNext() in D:\a\Synthesis\Synthesis\Mutagen.Bethesda.Synthesis\SynthesisPipeline.cs:line 378
Synthesis.Bethesda.Execution.CliUnsuccessfulRunException: Error running solution patcher
   at Synthesis.Bethesda.Execution.Patchers.Running.Solution.SolutionPatcherRunner.Run(RunSynthesisPatcher settings, CancellationToken cancel) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.Execution\Patchers\Running\Solution\SolutionPatcherRunner.cs:line 54
   at Synthesis.Bethesda.Execution.Patchers.Running.Solution.SolutionPatcherRun.Run(RunSynthesisPatcher settings, CancellationToken cancel) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.Execution\Patchers\Running\Solution\SolutionPatcherRun.cs:line 53
   at Synthesis.Bethesda.Execution.Running.Runner.RunAPatcher.Run(IGroupRun groupRun, PatcherPrepBundle prepBundle, CancellationToken cancellation, Nullable`1 sourcePath, RunParameters runParameters) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.Execution\Running\Runner\RunAPatcher.cs:line 81
   at Synthesis.Bethesda.Execution.Running.Runner.RunSomePatchers.Run(IGroupRun groupRun, PatcherPrepBundle[] patchers, CancellationToken cancellation, Nullable`1 sourcePath, RunParameters runParameters) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.Execution\Running\Runner\RunSomePatchers.cs:line 36
   at Synthesis.Bethesda.Execution.Running.Runner.RunAGroup.Run(IGroupRun groupRun, CancellationToken cancellation, DirectoryPath outputDir, RunParameters runParameters, Nullable`1 sourcePath) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.Execution\Running\Runner\RunAGroup.cs:line 59
   at Synthesis.Bethesda.Execution.Running.Runner.RunAllGroups.Run(IGroupRun[] groups, CancellationToken cancellation, DirectoryPath outputDir, RunParameters runParameters, Nullable`1 sourcePath) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.Execution\Running\Runner\RunAllGroups.cs:line 36
   at Synthesis.Bethesda.Execution.Running.Runner.ExecuteRun.Run(IGroupRun[] groups, CancellationToken cancellation, DirectoryPath outputDir, RunParameters runParameters, Nullable`1 sourcePath) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.Execution\Running\Runner\ExecuteRun.cs:line 54
   at Synthesis.Bethesda.GUI.Services.Profile.Running.ExecuteGuiRun.Run(IEnumerable`1 groupRuns, PersistenceMode persistenceMode, Boolean localize, Boolean utf8InEmbeddedStrings, Language targetLanguage, CancellationToken cancel) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.GUI\Services\Profile\Running\ExecuteGuiRun.cs:line 47
   at Synthesis.Bethesda.GUI.ViewModels.Profiles.Running.RunVm.<Run>b__39_1(Unit _) in D:\a\Synthesis\Synthesis\Synthesis.Bethesda.GUI\ViewModels\Profiles\Running\RunVm.cs:line 200

```

**Comment by RunicSkye** [comment]
Is fixing the issue just a matter of popping open Perks.cs and changing the numbers to match? Although I'm not quite sure where to get the updated numbers from... Like, I can see them in xedit, but even with a compatible version of the mod, the formIDs don't seem to line up?

**Comment by Drunkender** [comment]
https://github.com/tr4wzified/KnowYourArmorPatcher/commit/7417fa9711b046d21c97f642738c53cba63e01fe

^Should have fixed the 9.2.0 incompatibility.

**Comment by tr4wzified** [comment]
Yeah, sorry for not responding earlier, but that commit should've handled it. Let me know if any of you are still having issues.
